fedimint-api/src/tiered_multi.rs

```diff
@@ -141,19 +142,42 @@ where
 }
 
 impl TieredMulti<()> {
-    // TODO: move somewhere else?
-    pub fn represent_amount<K>(mut amount: Amount, tiers: &Tiered<K>) -> TieredMulti<()> {
-        let coins = tiers
-            .tiers()
-            .rev()
-            .map(|&amount_tier| {
-                let res = amount / amount_tier;
-                amount %= amount_tier;
-                (amount_tier, vec![(); res as usize])
-            })
-            .collect();
+    /// Determines the denominations to use when representing an amount
+    ///
+    /// Algorithm tries to leave the user with a target number of `denomination_sets` starting
+    /// at the lowest denomination.  `self` gives the denominations that the user already has.
+    pub fn represent_amount<K, V>(
+        mut amount: Amount,
+        current_denominations: &TieredMulti<V>,
+        tiers: &Tiered<K>,
+        denomination_sets: u16,
+    ) -> TieredMulti<()> {
```

Comment:
I agree with @dpc that this fn should probably just return `Tiered<usize>`. This will mean having to do some flat-mapping down the line, but that seems ok. It's not too bad as-is though since zero-sized-type `Vec`s are just 3*`usize` afaik and don't allocate, if the compiler is smart it will optimize it to one `usize`.